app/grandchallenge/discussion_forums/views.py

```diff
@@ -0,0 +1,92 @@
+from django.utils.functional import cached_property
+from django.views.generic import CreateView, DeleteView, DetailView, ListView
+
+from grandchallenge.core.guardian import (
+    ObjectPermissionRequiredMixin,
+    PermissionListMixin,
+)
+from grandchallenge.discussion_forums.forms import TopicForm
+from grandchallenge.discussion_forums.models import Topic, TopicTypeChoices
+from grandchallenge.subdomains.utils import reverse
+
+
+class TopicListView(PermissionListMixin, ListView):
+    model = Topic
+    permission_required = "discussion_forums.view_topic"
+    queryset = Topic.objects.select_related("forum")
+
+    @cached_property
+    def forum(self):
+        return self.request.challenge.discussion_forum
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        return queryset.filter(forum=self.forum)
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update(
+            {
+                "announcements": self.object_list.filter(
+                    type=TopicTypeChoices.ANNOUNCE
+                ),
+                "default_topics": self.object_list.exclude(
+                    type=TopicTypeChoices.ANNOUNCE
+                ),
+                "forum": self.forum,
+            }
+        )
+        return context
+
+
+class TopicCreate(ObjectPermissionRequiredMixin, CreateView):
+    model = Topic
+    permission_required = "discussion_forums.create_forum_topic"
+    raise_exception = True
+    form_class = TopicForm
+
+    @cached_property
+    def forum(self):
+        return self.request.challenge.discussion_forum
+
+    def get_permission_object(self):
+        return self.forum
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context.update({"forum": self.forum})
+        return context
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({"user": self.request.user, "forum": self.forum})
+        return kwargs
+
+
+class TopicDetail(ObjectPermissionRequiredMixin, DetailView):
+    model = Topic
+    permission_required = "discussion_forums.view_topic"
+    raise_exception = True
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context.update({"forum": self.object.forum})
+        return context
+
+
+class TopicDelete(ObjectPermissionRequiredMixin, DeleteView):
+    model = Topic
+    permission_required = "discussion_forums.delete_topic"
+    raise_exception = True
+    success_message = "Successfully deleted topic."
```

Comment:
I again took the current implementation as guidance. For the `TopicDelete` view this means that we don't block deletion of topics with posts and that deleting a topic also deletes all of its posts. This is also how the `ForeignKey` `on_delete` is currently set-up on the DB level. The permission to do this is reserved to challenge admins. 

My initial thought here as the the creator should also get the right to delete their own topics, but that would essentially give them the permission to delete other users posts to their topic, so I removed the permission from the creator.

Reply:
I think that users should be able to delete their own posts. If there are then no posts for a topic then it should also be deleted.

Reply:
Yes, posts can be deleted by the creator and if the last post of a topic is deleted, so is the topic. But what I meant here is whether the creator of a topic should be able to delete the topic. Currently, that is not possbile, and I've kept it that way since that would allow them to delete other user's posts.